### PR TITLE
Remove the need to hardcode the STATIC_CRON_TOKEN in the docker compose

### PR DESCRIPTION
**Pull request by mamoit**
This PR removes the need to hard code the `STATIC_CRON_TOKEN` value in the docker-compose files, making them free of secrets.

Documentation for the docker-compose feature that makes this change possible is [here](https://docs.docker.com/reference/compose-file/interpolation/).

I've tested it locally but since I'm just now setting up firefly-iii for the first time I may be missing some corner case or exotic use case that I'm not yet aware of.

### CODE REVIEW

**Comment by JC5** [comment]
Mmm, you could be right, nice catch. Did you test this already?

**Comment by mamoit** [comment]
Hi, I did a quick test before committing the PR.
Now I triple validated the change and expanded the PR to include a fix for the suspicion I raised on my previous comment.

Regarding 40a461a (Timezone env var expansion issue):
Without this fix, since I'm running a bare alpine VM without an env var `TZ` defined I got the following error:
```
WARN[0000] The "TZ" variable is not set. Defaulting to a blank string.
```
and `/etc/localtime` is a symlink to `/usr/share/zoneinfo/` instead of a link to `/usr/share/zoneinfo/Europe/Lisbon` (in my case).

With the fix, the symlink is perfect since the env var is expanded only inside the container.

Regarding bb82b72:
The main change here is to not pipe the result of the `ln` command into the echo, which I think is useless (please correct me if I'm wrong).
I also moved the `&&` to the end of the lines, but that is entirely personal preference for readability and what I've seen most people use throughout my career, feel free to request a revert and I'll do it straight away.

**Comment by mamoit** [comment]
Note to self that [the doc](https://github.com/firefly-iii/docs/blob/main/docs/docs/how-to/firefly-iii/advanced/cron.md#docker-compose) will need to be altered for consistency if this PR is merged.
I'm not sure if there is somewhere else where this change needs to be propagated to.

File: docker-compose-importer.yml [hunk 1]
```diff
@@ -47,19 +47,19 @@ services:
...
   cron:
     #
-    # To make this work, set STATIC_CRON_TOKEN in your .env file or as an environment variable and replace PLEASE_REPLACE_WITH_32_CHAR_CODE below
+    # To make this work, set STATIC_CRON_TOKEN in your .env file or as an environment variable
     # The STATIC_CRON_TOKEN must be *exactly* 32 characters long
     # Use this URL for inspiration: https://www.random.org/strings/?num=1&len=32&digits=on&upperalpha=on&loweralpha=on&unique=on&format=html&rnd=new
     #
     image: alpine
     container_name: firefly_iii_cron
     restart: always
```
Comment (JC5) [inline docker-compose-importer.yml:56]: I just realized the cron entry still needs a `env_file: .env` to function.

**Review by JC5 — CHANGES_REQUESTED** [review]

File: docker-compose-importer.yml (hunk 1 above)
Comment (mamoit) [reply docker-compose-importer.yml:56]: @PrplHaz4 as @JC5 said indeed that's possible, but you would be repeating the secrets in a file that is supposed to be able to be shareable and not have any sensitive information.
Ideally we'd use [secrets](https://docs.docker.com/compose/how-tos/use-secrets/), but that's for another PR.

@JC5 I just committed the inclusion of the `env_file` for the `docker-compose-importer.yaml`.
I was thinking that it may be a good idea to include the `docker-compose.yaml` from the `docker-compose-importer.yaml` to keep the docker compose DRY and avoid splitbrain issues like this lack of `env_file` in one of the `docker-compose` files.
I can do another PR after this one to propose this change and avoid deviating too much from the original purpose of this PR.

**Comment by JC5** [comment]
I saw it crash a few times because `ln -s` fails. This was a basic test to see if the upgrade from 6.2 to 6.3 works. But still:

```
firefly_iii_cron  | OK: 8 MiB in 17 packages
firefly_iii_cron  | ln: /etc/localtime: File exists
firefly_iii_cron exited with code 1
```

Surprisingly, it worked as expected in a later run. 

```
firefly_iii_cron  | (1/1) Installing tzdata (2025b-r0)
firefly_iii_cron  | OK: 8 MiB in 17 packages
firefly_iii_cron  | crond: crond (busybox 1.37.0) started, log level 8
```

So not yet sure what it could be. It may be an idea to use `-f`, what do you think?

**Comment by mamoit** [comment]
The fact that it fails is a good thing since before it would silently fail 😅
Can you tell me how to reproduce the crash?
Was it on a clean environment?

I'm afk right now, but my first suspicion is that on a container restart (not recreation) the link is already there, and since the command runs on every start the `ln` fails.

I can dive into it in a couple of hours.

**Comment by JC5** [comment]
No rush. This was a clean environment in between docker compose up/down. Will try to reproduce it, no rush.

**Comment by JC5** [comment]
Yeah that makes sense. I've merged the changes, thanks for your time and an excellent discussion! I will match the docs with the new command.

**Review by JC5 — APPROVED** [review]

**Comment by mamoit** [comment]
Happy to help out 🙂
I'm still trying out firefly-iii, if I stick to it you can expect more spam from me.
